pangeo_forge_esgf/dynamic_kwargs.py

```diff
@@ -96,9 +97,11 @@ async def response_data_processing(
     session: aiohttp.ClientSession,
     response_data: List[Dict[str, str]],
     iid: str,
+    ssl: ssl.SSLContext = None,
```

Comment:
For now this is great, but adding all these input arguments is a bit of a code smell to me. I think we should aim to set up a 'Connection' or 'Client' object once and then pass this around the different logic that needs to make a request?

Reply:
Agreed!